src/plans/details/components/PlanMigrationTypeLabel/PlanMigrationTypeLabel.tsx

```diff
@@ -0,0 +1,33 @@
+import type { FC } from 'react';
+import { MigrationTypeValue } from 'src/plans/create/steps/migration-type/constants';
+
+import { Button, ButtonVariant, Label, Popover } from '@patternfly/react-core';
+import { useForkliftTranslation } from '@utils/i18n';
+
+import { bodyContent, typeLabel } from './utils/utils';
+
+import './PlanMigrationTypeLabel.scss';
+
+type PlanMigrationTypeLabelProps = {
+  migrationType: MigrationTypeValue;
+};
+
+const PlanMigrationTypeLabel: FC<PlanMigrationTypeLabelProps> = ({ migrationType }) => {
+  const { t } = useForkliftTranslation();
+
+  return (
+    <Popover
+      headerContent={t('{{type}} migration', { type: typeLabel(migrationType) })}
+      bodyContent={bodyContent(migrationType)}
+      triggerAction="hover"
+    >
+      <Button className="forklift-plan-migration-type-label" variant={ButtonVariant.plain} isInline>
+        <Label isCompact color={migrationType === MigrationTypeValue.Cold ? 'blue' : 'orange'}>
+          {t(typeLabel(migrationType))}
```

Comment:
t func doesn't work on dynamic values in a direct manner, for this to work it should have something like:
`t('{{label}}', { label: typeLabel(migrationType) })`. or you can go and add translations in the func itself